expr/interval_year_to_month.go

```diff
@@ -0,0 +1,36 @@
+package expr
+
+import (
+	"github.com/substrait-io/substrait-go/proto"
+	"github.com/substrait-io/substrait-go/types"
+)
+
+type intervalYearMonthVal struct {
+	years  int32
+	months int32
+}
+
+// NewIntervalLiteralUptoMonth creates an interval literal which allows setting only year and month
+// arguments: nullable property (n), years and months
+func NewIntervalLiteralUptoMonth(n types.Nullability, years int32, months int32) Literal {
+	intervalCompoundType := types.NewIntervalYearToMonthType().WithNullability(n)
+	intervalPartsVal := &intervalYearMonthVal{
+		years:  years,
+		months: months,
+	}
```

Comment:
this structure represents value. proto carries type related information too (e.g. [Precision](https://github.com/substrait-io/substrait/blob/28025cbaa8dc3c65b736d8a68fa7070c465fb494/proto/substrait/algebra.proto#L924)). My intention is to construct value by value related fields.
We have another object to represent type so wanted to keep it separate so that separation of value/type remain clear

Reply:
Resolving this comment, since I assume (based on other comments) that we agree to not use Proto to construct literal and have a separate struct which implements literal is ok